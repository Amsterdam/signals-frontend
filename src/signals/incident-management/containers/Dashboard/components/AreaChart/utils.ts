// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam
import { Direction } from './ComparisonRate'
import type { ComparisonRateType } from './ComparisonRate'
import type { AreaChartValue as Value, Today } from '../../charts/types'

export const formatData = (values: Value[]): Value[] => {
  return values.map((value, i) =>
    i === values.length - 1
      ? {
          ...value,
          date: `${value.date} 12:00`,
          image: '/assets/images/area-chart-tooltip.svg',
        }
      : { ...value, date: `${value.date} 12:00` }
  )
}

export const getToday = (): Today => {
  const today = new Date()

  return {
    year: today.getFullYear(),
    // Vega lite does not work with zero indexed months
    month: today.getMonth() + 1,
    date: today.getDate(),
    hours: 12,
    minutes: 0o0,
  }
}

export const getPercentage = (values: Value[]): ComparisonRateType => {
  const thisWeek = values.reduce((total, value) => total + value.amount, 0)
  const lastWeek = values.reduce(
    (total, value) => total + value.amount_week_earlier,
    0
  )

  return {
    direction: thisWeek < lastWeek ? Direction.DOWN : Direction.UP,
    percentage: Math.abs(Math.round((thisWeek / lastWeek - 1) * 100)),
  }
}
