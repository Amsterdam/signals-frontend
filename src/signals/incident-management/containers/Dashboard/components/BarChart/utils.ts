// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam
import configuration from 'shared/services/configuration/configuration'

import type { RawData } from './types'
import { constants } from '../../charts'
import type { BarChartValue } from '../../charts'

export const getQueryList = (queryString: string) => {
  const endpoint = configuration.INCIDENT_PRIVATE_ENDPOINT
  return constants.statusList.map((status) => {
    const params = queryString
      ? `${queryString}&status=${status.query}`
      : `status=${status.query}`
    return `${endpoint}stats/total?${params}`
  })
}

export const formatData = (rawData: RawData[]): BarChartValue[] => {
  return rawData.map((value, i) => {
    return {
      status: constants.statusList[i].label,
      nrOfIncidents: value.total,
    }
  })
}

export const getTotalNrOfIncidents = (rawData: RawData[]): number => {
  return rawData.reduce(
    (aggregatedTotal, value) => aggregatedTotal + value.total,
    0
  )
}
