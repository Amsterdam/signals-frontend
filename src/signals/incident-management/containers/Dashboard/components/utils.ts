// SPDX-License-Identifier: MPL-2.0
// Copyright (C) 2023 Gemeente Amsterdam
import type { IncidentCount } from '../charts/types'

export const getMaxDomain = (values: IncidentCount[]) =>
  Math.max(...values.map((v) => v.amount)) * 1.2
