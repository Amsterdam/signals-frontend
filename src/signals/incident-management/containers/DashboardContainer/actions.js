import {
  REQUEST_DASHBOARD,
  REQUEST_DASHBOARD_SUCCESS
} from './constants';

export function requestDashboard() {
  return {
    type: REQUEST_DASHBOARD
  };
}

export function requestDashboardSuccess(results) {
  return {
    type: REQUEST_DASHBOARD_SUCCESS,
    payload: results
  };
}
