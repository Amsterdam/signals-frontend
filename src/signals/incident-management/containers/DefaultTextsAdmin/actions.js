import {
  FETCH_DEFAULT_TEXTS, FETCH_DEFAULT_TEXTS_SUCCESS, FETCH_DEFAULT_TEXTS_ERROR,
  STORE_DEFAULT_TEXTS, STORE_DEFAULT_TEXTS_SUCCESS, STORE_DEFAULT_TEXTS_ERROR,
  ORDER_DEFAULT_TEXTS,
} from './constants';

export function fetchDefaultTexts(payload) {
  return {
    type: FETCH_DEFAULT_TEXTS,
    payload,
  };
}

export function fetchDefaultTextsSuccess(payload) {
  return {
    type: FETCH_DEFAULT_TEXTS_SUCCESS,
    payload,
  };
}

export function fetchDefaultTextsError() {
  return {
    type: FETCH_DEFAULT_TEXTS_ERROR,
  };
}

export function storeDefaultTexts(payload) {
  return {
    type: STORE_DEFAULT_TEXTS,
    payload,
  };
}

export function storeDefaultTextsSuccess(payload) {
  return {
    type: STORE_DEFAULT_TEXTS_SUCCESS,
    payload,
  };
}

export function storeDefaultTextsError() {
  return {
    type: STORE_DEFAULT_TEXTS_ERROR,
  };
}

export function orderDefaultTexts(payload) {
  return {
    type: ORDER_DEFAULT_TEXTS,
    payload,
  };
}
