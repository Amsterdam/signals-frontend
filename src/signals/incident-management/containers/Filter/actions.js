import {
  SAVE_FILTER,
  SAVE_FILTER_FAILED,
  SAVE_FILTER_SUCCESS,
  UPDATE_FILTER,
  UPDATE_FILTER_SUCCESS,
  UPDATE_FILTER_FAILED,
  GET_FILTERS,
  GET_FILTERS_SUCCESS,
  GET_FILTERS_FAILED,
  CLEAR_FILTER,
} from './constants';

export const filterSaved = (payload) => ({
  type: SAVE_FILTER,
  payload,
});

export const filterSaveFailed = (payload) => ({
  type: SAVE_FILTER_FAILED,
  payload,
});

export const filterSaveSuccess = (payload) => ({
  type: SAVE_FILTER_SUCCESS,
  payload,
});

export const filterUpdated = (payload) => ({
  type: UPDATE_FILTER,
  payload,
});

export const filterUpdatedSuccess = (payload) => ({
  type: UPDATE_FILTER_SUCCESS,
  payload,
});

export const filterUpdatedFailed = (payload) => ({
  type: UPDATE_FILTER_FAILED,
  payload,
});

export const getFilters = () => ({
  type: GET_FILTERS
});

export const getFiltersSuccess = (payload) => ({
  type: GET_FILTERS_SUCCESS,
  payload,
});

export const getFiltersFailed = (payload) => ({
  type: GET_FILTERS_FAILED,
  payload,
});

export const filterCleared = () => ({
  type: CLEAR_FILTER,
  payload: {},
});
