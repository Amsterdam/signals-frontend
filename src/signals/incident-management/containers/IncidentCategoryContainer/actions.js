import {
  REQUEST_CATEGORY_UPDATE,
  REQUEST_CATEGORY_UPDATE_SUCCESS,
  REQUEST_CATEGORY_UPDATE_ERROR
}
  from './constants';

export function requestCategoryUpdate(category) {
  return {
    type: REQUEST_CATEGORY_UPDATE,
    payload: category
  };
}

export function requestCategoryUpdateSuccess(category) {
  return {
    type: REQUEST_CATEGORY_UPDATE_SUCCESS,
    payload: category
  };
}

export function requestCategoryUpdateError(message) {
  return {
    type: REQUEST_CATEGORY_UPDATE_ERROR,
    payload: message
  };
}
