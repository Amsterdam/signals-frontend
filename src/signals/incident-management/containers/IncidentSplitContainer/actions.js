import {
  SPLIT_INCIDENT,
  SPLIT_INCIDENT_SUCCESS,
  SPLIT_INCIDENT_ERROR,
} from './constants';

export function splitIncident(payload) {
  return {
    type: SPLIT_INCIDENT,
    payload,
  };
}

export function splitIncidentSuccess(result) {
  return {
    type: SPLIT_INCIDENT_SUCCESS,
    payload: result,
  };
}

export function splitIncidentError(error) {
  return {
    type: SPLIT_INCIDENT_ERROR,
    payload: error,
  };
}
