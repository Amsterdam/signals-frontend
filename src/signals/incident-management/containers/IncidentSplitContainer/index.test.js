import React from 'react';
import { render } from '@testing-library/react';
import * as reactRouterDom from 'react-router-dom';
import { store, withAppContext } from 'test/utils';
import incidentJson from 'utils/__tests__/fixtures/incident.json';
import categoriesPrivate from 'utils/__tests__/fixtures/categories_private.json';
import { fetchCategoriesSuccess } from 'models/categories/actions';
import { requestIncidentSuccess } from 'models/incident/actions';

import { IncidentSplitContainer } from './index';

jest.mock('react-router-dom', () => ({
  __esModule: true,
  ...jest.requireActual('react-router-dom'),
}));

store.dispatch(requestIncidentSuccess(incidentJson));
store.dispatch(fetchCategoriesSuccess(categoriesPrivate));

jest.spyOn(reactRouterDom, 'useParams').mockImplementation(() => ({
  id: '42',
}));

// mocking a deeply nested component to prevent having to mock
// multiple data providers and child components
jest.mock('../../components/FieldControlWrapper', () => ({
  __esModule: true,
  default: () => (<span />),
}));

describe('<IncidentSplitContainer />', () => {
  it('should render correctly', async () => {
    const props = {
      onSplitIncident: jest.fn(),
      onGoBack: jest.fn(),
    };

    fetch.mockResponseOnce(JSON.stringify(incidentJson));

    const { queryByTestId, queryAllByTestId, findByTestId } = render(
      withAppContext(<IncidentSplitContainer {...props} />)
    );

    expect(queryByTestId('loadingIndicator')).toBeInTheDocument();

    await findByTestId('incidentSplit');

    expect(queryByTestId('loadingIndicator')).not.toBeInTheDocument();

    expect(queryAllByTestId('incidentPartTitle')[0]).toHaveTextContent(
      /^Deelmelding 1$/
    );
    expect(queryAllByTestId('incidentPartTitle')[1]).toHaveTextContent(
      /^Deelmelding 2$/
    );
  });
});
