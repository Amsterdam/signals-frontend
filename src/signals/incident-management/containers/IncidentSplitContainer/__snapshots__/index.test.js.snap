// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IncidentSplitContainer /> rendering should render correctly 1`] = `
<div
  className="incident-split-container"
>
  <div
    className="row"
  >
    <div
      className="col-8"
    >
      <SplitForm
        attachments={Array []}
        handleCancel={[Function]}
        handleSubmit={[Function]}
        incident={Object {}}
        isVisible={false}
        priorityList={
          Array [
            Object {
              "key": "normal",
              "value": "Normaal",
            },
            Object {
              "key": "high",
              "value": "Hoog",
            },
          ]
        }
        subcategories={Array []}
      />
    </div>
    <div
      className="col-4"
    >
      <SplitDetail
        incident={Object {}}
        stadsdeelList={
          Array [
            Object {
              "key": "A",
              "value": "Centrum",
            },
            Object {
              "key": "B",
              "value": "Westpoort",
            },
            Object {
              "key": "E",
              "value": "West",
            },
            Object {
              "key": "M",
              "value": "Oost",
            },
            Object {
              "key": "N",
              "value": "Noord",
            },
            Object {
              "key": "T",
              "value": "Zuidoost",
            },
            Object {
              "key": "K",
              "value": "Zuid",
            },
            Object {
              "key": "F",
              "value": "Nieuw-West",
            },
          ]
        }
      />
    </div>
  </div>
</div>
`;

exports[`<IncidentSplitContainer /> rendering should render loading indicator correctly 1`] = `
<div
  className="incident-split-container"
>
  <LoadingIndicator />
</div>
`;
